lib/modules/deployment/index.js

```diff
@@ -81,25 +81,20 @@ class DeployManager {
               contractDeploys[className].push(deploy);
             });
 
-            try {
```

Comment:
Can you try to put a circular dependency on your contracts (just need to put ContractA in ContractB's args and vice-versa) and see if it throws or if it calls back. Just wondering if that's why we had a try catch